.github/actions/setup-miniconda/README.md

```diff
@@ -0,0 +1,21 @@
+# setup-miniconda
+
+Sets up miniconda in your `${RUNNER_TEMP}` environment and gives you the `${CONDA_RUN}` environment variable so you don't have to worry about polluting non-empeheral runners anymore
+
+## Supported platforms
+
+- macOS-ARM64
+- macOS-X64
+
+## Usage
+
+```yaml
+      - name: Setup miniconda
+        # You could potentially lock this down to a specific hash as well
+        uses: pytorch/test-infra/.github/actions/setup-miniconda@main
```

Comment:
nit: do we want to put `macos` in the name of the action to clarify that it only works for macos?

Reply:
I anticipate that we'll add support for other platform since it'll be as simple as adding more case statements.

So in an ideal world there'll be a follow up to this to expand platform support